src/plugins/intel_cpu/src/nodes/fullyconnected.h

```diff
@@ -52,76 +70,42 @@ class FullyConnected : public Node {
-        this->weightsNonTransposed = weightsNonTransposed;
+        this->attrs.weightsNonTransposed = weightsNonTransposed;
     }
 
     void fuseDecompressionMultiply(const MemoryCPtr& memory);
     void fuseDecompressionSubtract(const MemoryCPtr& memory);
+    // defines which of the inputs are constant
+    constexpr static int mask() {
+        return 0 + (1 << 1);
+    }
 
 private:
-    void createDescriptorInternal(const dnnl::memory::desc &inputDesc,
-                                  const dnnl::memory::desc &outputDesc);
-
-    VectorDims makeDummyInputDims() const;
-    VectorDims makeDummyOutputDims(const VectorDims& inDims) const;
-
-    VectorDims inDims;
-    VectorDims outDims;
-
-    void setPostOps(dnnl::primitive_attr &attr, const VectorDims &dims);
-
-    bool withBiases = false;
-
     std::string errorPrefix;
     static const size_t DATA_ID = 0;
     static const size_t WEIGHTS_ID = 1;
     static const size_t BIAS_ID = 2;
-    dnnl::memory::data_type outputDataType = dnnl::memory::data_type::undef;
-
-    using executorPtr = std::shared_ptr<DnnlExecutor>;
-    executorPtr execPtr = nullptr;
-    bool useConv1x1 = false;
-    impl_desc_type implementationTypeIP = impl_desc_type::unknown;
-    MemoryDescPtr weightDescIP;
-    dnnl::primitive_attr attr;
-
-    static dnnl::convolution_forward::primitive_desc
-    createDescriptorInternalForConv(DnnlMemoryDescCPtr inputDescPtr,
-                                    DnnlMemoryDescCPtr weightDescPtr,
-                                    DnnlMemoryDescCPtr biasDescPtr,
-                                    DnnlMemoryDescCPtr outputDescPtr,
-                                    const dnnl::primitive_attr& attr,
-                                    const dnnl::engine& engine);
-
-    bool canBeExecutedInConv1x1() const;
+
+    ExecutorPtr executor = nullptr;
+
+    ExecutorPtr createExecutor();
     void fuseDecompressionConstant(const MemoryCPtr& memory, MemoryCPtr& decompressionValuesPtr);
 
-    // sparse weights
-    bool useSparseWeights = false;
-    float minSparseRate = 1.f;
-    float weiSparseRate = 0.f;
-    bool useSparseWeightsDecompression();
-    VectorDims expectedBiasDims {};
-    bool useMlas = false;
-#ifdef OV_CPU_WITH_MLAS
-    int64_t M, N, K;
-    MemoryPtr mlasPackedPtr = nullptr;
-    void executeMLAS();
-    void prepackMLASWeight();
-#endif
-#if defined(OV_CPU_WITH_ACL)
-    void prepareWeightsUsingDummyShape();
-#endif
-    bool useWeightsDecompressionImpl = false;
-    MemoryCPtr decompressionSubtractPtr = nullptr;
-    MemoryCPtr decompressionMultiplyPtr = nullptr;
-
-    // FC with transposed weights
-    bool weightsNonTransposed = false;
-    DnnlMemoryDescPtr makeTransposedWeightDescriptor(DnnlMemoryDescPtr desc);
+    FCAttrs attrs;
+    PostOps postOps;
+    ExecutorFactoryNewPtr<FCAttrs, node::FullyConnected> factory;
+    MemoryArgs memory;
+    MemoryDescArgs descriptors;
+    MemoryPtr emptyMemory;
```

Comment:
This one I actually wanted to discuss. The idea is to use some empty memory object for biases, in case the node does not have bias input (to generalize all the primitive / executor creation logic). We could have it as a static member, or maybe even inside the graph context (since the same object will basically fit any node). What do you think?

Reply:
What if simply don't insert any entries to the maps of memory objects and descriptor vectors if there is no such input at all?

Reply:
This requires having additional checks later in a pipeline in multiple places, which are not necessary, since all the backends (mlas, oneDNN, acl) accepts nullptr / empty memory as a bias input. So, we can just access bias memory and pass it to the appropriate backend, even if it is empty.
As an example, you can compare `MlasGemmExecutor::execute` (no bias check) with `FullyConnected::executeMLAS` (additional bias check) from master.

Reply:
Thank you for making this clear. Now I got the point. But it raised yet another question, please check it out: https://github.com/openvinotoolkit/openvino/pull/20718#discussion_r1457681439